src/pathfinder/ui/core.cljs

```diff
@@ -1,25 +1,50 @@
 (ns ^:figwheel-always pathfinder.ui.core
+  (:require-macros [cljs.core.async.macros :as am])
     (:require[om.core :as om :include-macros true]
-             [om.dom :as dom :include-macros true]))
+             [om.dom :as dom :include-macros true]
+             [pathfinder.ui.render :as render]
+             [cljs.core.async :as a]
+             [ajax.core :refer [GET]]))
 
 (enable-console-print!)
 
 (println "Edits to this text should show up in your developer console.")
 
-;; define your app data so that it doesn't get over-written on reload
+(defn update-state
+  [app display-text]
+  (let [new-state (swap! (:state app) assoc :search-results display-text)]
+    (render/request-render app)))
 
-(defonce app-state (atom {:text "Hello world! This has come from js generated by cljs"}))
+(defn query-server
+  [app val]
+  (GET (str (:server-uri app) "/projects?q=" val)
```

Comment:
Do we need the server-uri? Should be enough to query relatively. We should get a handle on being restful sooner rather than later and provide this uri to the page.